tests/test_esss_fix_format.py

```diff
@@ -143,6 +143,23 @@ def check_output(cmd, *args, **kwargs):
     ]
 
 
+def test_input_invalid_codec(tmpdir):
+    """Display error summary when we fail to open a file"""
+    filename = tmpdir.join('test.py')
+    filename.write(u'hello world'.encode('UTF-16'), 'wb')
+    output = run([str(filename)], expected_exit=1)
+    output.fnmatch_lines(str(filename) + ': ERROR (Unicode*')
+    output.fnmatch_lines('*== ERRORS ==*')
+    output.fnmatch_lines(str(filename) + ': ERROR (Unicode*')
+
+
+def test_empty_file(tmpdir):
+    """Ensure files with a single empty line do not raise an error"""
```

Comment:
No, that was because we were missing the `u` for unicode on the string literals for the eol strings in the `_split_lines_and_original_eol`: for an empty file, all that was added to the final contents was that `bytes` string, which caused an error when trying to write to an file opened with `io.open` in text mode.